mempool/src/error/ban_score.rs

```diff
@@ -357,3 +354,18 @@ impl MempoolBanScore for accounting::Error {
         }
     }
 }
+
+impl MempoolBanScore for IOPolicyError {
+    fn mempool_ban_score(&self) -> u32 {
+        match self {
+            IOPolicyError::InvalidInputTypeInReward => 100,
```

Comment:
@iljakuklic Can you please double-check that these values make sense in the context of the mempool?

Reply:
> @iljakuklic Can you please double-check that these values make sense in the context of the mempool?

Just quickly skimming over it:
* The `Invalid*Type*` arms look fine
* So does `ProduceBlockInTx` and `ConstrainedAmountOverflow` if I understand them correctly.
* I am not convinced by `MultiplePoolCreated`, `MultipleDelegationCreated` since these look like something that depends on the current chain state.
* I don't know about `TimelockRequirementNotSatisfied`, have not looked in the code too deeply to understand that. I suspect that one may be problematic too.

Please note this is based on a rather superficial reading of these errors and there may be aspects that I did not understand correctly.

In general, a rule of thumb in mempool is:
* If the error depends on the current state of the blockchain, ban score should be 0. This is because in a distributed system, it has to be tolerated for nodes to be temporarily out of sync.
* If the error signifies the transaction is invalid no matter which block it is included in, give it a ban score 100.

Reply:
`MultiplePoolCreated` and `MultipleDelegationCreated` do not depend on a state. They indicate that transaction has > 1 `CreateStakePool` or `CreateDelegationId` outputs which is invalid since the id won't be unique.

`TimelockRequirementNotSatisfied` also should be fine as it shows that not enough coins were locked in the outputs.

Reply:
Makes sense, It's all fine then